HW3/P4/median_filter.cl

```diff
@@ -31,4 +80,9 @@ median_3x3(__global __read_only float *in_values,
 
     // Each thread in the valid region (x < w, y < h) should write
     // back its 3x3 neighborhood median.
+    if ((x < w) && (y < h)) {
+        out_values[x + y*w] = median9(buffer[lx + ly*buf_w] , buffer[lx + 1 + ly*buf_w], buffer[lx + 2 + ly*buf_w],
```

Comment:
Suggestion: Instead of accessing values in the buffer using lx, ly, you can access them using buf_x, buf_y (easier to understand).